pandas/_libs/tslibs/timestamps.pyx

```diff
@@ -409,7 +410,12 @@ class Timestamp(_Timestamp):
                           " tz parameter will raise in the future. Use"
                           " tz_convert instead.", FutureWarning)
 
-        ts = convert_to_tsobject(ts_input, tz, unit, 0, 0, nanosecond or 0)
+        try:
+            ts = convert_to_tsobject(ts_input, tz, unit, 0, 0, nanosecond or 0)
+        except OverflowError:
```

Comment:
can you update the doc-string of convert_ot_tsobject. Why are we not catching the OverflowError *inside*?

Reply:
I put this here because I wanted to do this error-catching in python-space instead of c-space so that convert_to_tsobject could have better optimizations, but it looks like convert_to_tsobject already has some raising cases, so this can be moved there.  will update.